src/m_misc.c

```diff
@@ -2258,20 +2258,18 @@ default_t defaults[] = {
     "1 to enable fullscreen mode"
   },
 
-  // [FG] precache all sound effects
   {
-    "precache_sounds",
-    (config_t *) &precache_sounds, NULL,
-    {1}, {0, 1}, number, ss_gen, wad_no,
-    "1 to precache all sound effects"
+    "snd_samplerate",
```

Comment:
The main reason why I made this value configurable is not that I want users to arbitrarily change this value. It's is because sometimes SDL2_Mixer cannot open the audio device with the desired sample rate, e.g. 44100 Hz vs 48000 Hz. However, the snd_samplerate value is used *before* the device is opened to calculate the "slice size". If we never feed back the actual sample rate into the config variable, then the sound device will *always* get opened with the wrong slice size based on the assumption of a 44100 Hz sample rate.